### PR TITLE
[cody/web] Papercuts: Hide delete & update history icon

### LINKED ISSUE #55746: Papercuts: Hide delete action for empty interactions / Update history icon
- Hide delete action for empty interactions
- Update history icon

### PR DESCRIPTION
**Pull request by st0nebraker**
Closes #55746 - Hides delete chat item or clear all chat history when there are no chat interactions. Updates chat history icon to match [VS Code's](https://github.com/sourcegraph/cody/pull/323)

### Empty chat history
Screenshot 2023-08-10 at 9 20 52 AM
Screenshot 2023-08-10 at 9 25 40 AM

### Filled chat history
Screenshot 2023-08-10 at 9 21 42 AM
Screenshot 2023-08-10 at 9 27 22 AM

## Test plan
- Manual test on cody sidebar & cody chat page: With chat history and without

### CODE REVIEW

File: client/web/src/cody/sidebar/CodySidebar.tsx [hunk 1]
```diff
@@ -92,15 +92,15 @@ export const CodySidebar: React.FC<CodySidebarProps> = ({ onClose }) => {
...
+                                <Icon aria-hidden={true} svgPath={mdiHistory} />
                             </Button>
                         </Tooltip>
                         <Tooltip content="Start a new chat">
                             <Button variant="icon" aria-label="Start a new chat" onClick={onReset}>
                                 <Icon aria-hidden={true} svgPath={mdiPlus} />
                             </Button>
                         </Tooltip>
-                        {showHistory && (
+                        {showHistory && transcriptHistory[0].interactions.length > 0 && (
```
Comment (thenamankumar) [inline client/web/src/cody/sidebar/CodySidebar.tsx:103]: "Clear all chats" depends on `transcriptHistory.length` and not on `transcriptHistory[0].interactions.length`

transcriptHistory stores the list of all chats. 

Need to change everywhere else too.

**Review by thenamankumar — CHANGES_REQUESTED** [review]

File: client/web/src/cody/sidebar/CodySidebar.tsx (hunk 1 above)
Comment (thenamankumar) [reply client/web/src/cody/sidebar/CodySidebar.tsx:104]: To handle that case the correct logic would be:

`showHistory && (transcriptHistory.length > 1  || !!transcriptHistory[0]?.interactions?.length)`

Comment (st0nebraker) [reply client/web/src/cody/sidebar/CodySidebar.tsx:104]: My question was about your original point & if there is another spot this change needs to touch?

Comment (thenamankumar) [reply client/web/src/cody/sidebar/CodySidebar.tsx]: Where all the "Clear all chats" logic is there. Delete history item logic is fine.

Comment (st0nebraker) [reply client/web/src/cody/sidebar/CodySidebar.tsx]: Thanks! Updated in [af6e49d](https://github.com/sourcegraph/sourcegraph/pull/55749/commits/af6e49d8ac332a04fe83f72dd867a1986e102574)

**Review by thenamankumar — APPROVED** [review]

**Pull request merged**